lib/livebook_web/live/session_live/app_settings_component.ex

```diff
@@ -170,14 +170,19 @@ defmodule LivebookWeb.SessionLive.AppSettingsComponent do
       Livebook.Session.set_app_settings(socket.assigns.session.pid, settings)
     end
 
-    if socket.assigns.live_action == :app_settings_and_launch do
-      {:noreply, deploy_app(socket, app_settings, deployed_app_slug)}
-    else
-      {:noreply, push_patch(socket, to: ~p"/sessions/#{socket.assigns.session.id}")}
+    case socket.assigns.context do
+      "preview" ->
+        {:noreply, preview_app(socket, app_settings, deployed_app_slug)}
+
+      redirect when is_binary(redirect) ->
+        {:noreply, push_patch(socket, to: ~p"/sessions/#{socket.assigns.session.id}/#{redirect}")}
```

Comment:
I was not sure if I used the same assign (context) or different ones, I went with the same and then conditionally handling it.